src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -441,6 +444,9 @@ public void Enqueue(IThreadPoolWorkItem callback, bool forceGlobal)
             if (null != tl)
             {
                 tl.workStealingQueue.LocalPush(callback);
+
+                if(ThreadPoolGlobals.wsqActive != ThreadPoolGlobals.wsqNowActive)
+                   Interlocked.Exchange(ref ThreadPoolGlobals.wsqActive, ThreadPoolGlobals.wsqNowActive);
```

Comment:
Set wsqActive to wsqNowActive, if and only if it was observed empty.

Using a Interlocked Exchange to guarantee a volatile write with Memory Barrier.  Seems reasonable, but I am not sure it is the preferred idiom.